### ISSUE TITLE
The project does not contain config "Debug" for building with Cmake

### ISSUE THREAD

**Issue by DylanWake** [opencv/opencv#21308]
##### System information (version)

- OpenCV => Opencv 4.5.2, Opencv-Contrib 4.5.2
- Operating System / Platform => Windows, AMD64, CUDA 11.5, GPU_ARCH: sm_86
- Compiler => MSVC 2019 (C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools) , NVCC

##### Detailed description

I have compiled the OpenCV project with Release configs, and it worked correctly. However, some methods like `cv::imread` requires the libraries of OpenCV to be in Debug mode in order to be working in Debug builds of my project.

I followed the steps in OpenCV Docs: https://docs.opencv.org/4.x/db/d05/tutorial_config_reference.html#tutorial_config_reference_general_debug

However, when I set the build config in Cmake to Debug, the following happened:
```
D:\OpenCV_CUDA>cmake --build "./build" --target INSTALL --config Debug
Microsoft (R) Build Engine version 16.11.1+3e40a09f8 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(436,
5): error MSB8013: This project doesn't contain the Configuration and Platform combination of Debug|x64. [D:\OpenCV_CUD
A\build\3rdparty\openjpeg\openjp2\libopenjp2.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(436,
5): error MSB8013: This project doesn't contain the Configuration and Platform combination of Debug|x64. [D:\OpenCV_CUD
A\build\3rdparty\ippiw\ippiw.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(436,
5): error MSB8013: This project doesn't contain the Configuration and Platform combination of Debug|x64. [D:\OpenCV_CUD
A\build\3rdparty\ittnotify\ittnotify.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(436,
5): error MSB8013: This project doesn't contain the Configuration and Platform combination of Debug|x64. [D:\OpenCV_CUD
A\build\3rdparty\zlib\zlib.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(436,
5): error MSB8013: This project doesn't contain the Configuration and Platform combination of Debug|x64. [D:\OpenCV_CUD
A\build\3rdparty\libjpeg-turbo\libjpeg-turbo.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(436,
5): error MSB8013: This project doesn't contain the Configuration and Platform combination of Debug|x64. [D:\OpenCV_CUD
A\build\3rdparty\libwebp\libwebp.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(436,
5): error MSB8013: This project doesn't contain the Configuration and Platform combination of Debug|x64. [D:\OpenCV_CUD
A\build\3rdparty\protobuf\libprotobuf.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(436,
5): error MSB8013: This project doesn't contain the Configuration and Platform combination of Debug|x64. [D:\OpenCV_CUD
A\build\3rdparty\quirc\quirc.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(436,
5): error MSB8013: This project doesn't contain the Configuration and Platform combination of Debug|x64. [D:\OpenCV_CUD
A\build\modules\ade.vcxproj] 
```
       

##### Steps to reproduce

1: clone OpenCV and OpenCV-Contrib repositories to your computer
2: set the Cmake configurations like following:
[configuration.txt](https://github.com/opencv/opencv/files/7759043/configuration.txt)
3: run build with:
```
cmake --build "./build" --target INSTALL --config Debug
```
    

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by alalek** [comment]
[Can't reproduce](http://pullrequest.opencv.org/buildbot/builders/master_noOCL_noICV_noSSE-win64-vc14-debug).

---

> 4.5.2

Outdated release. Follow the template:

> I updated to latest OpenCV version and the issue is still there

---

Try without CUDA stuff.

**Comment by DylanWake** [comment]
Issue resolved by compiling OpenCV with version 4.5.4
